src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/UserEntity.kt

```diff
@@ -368,6 +368,13 @@ enum class UserRole(val service: ServiceName, val cas1ApiValue: ApprovedPremises
     fun valueOf(apiValue: ApprovedPremisesUserRole) = UserRole.entries.first { it.cas1ApiValue == apiValue }
 
     fun getAllRolesForPermission(permission: UserPermission) = UserRole.values().filter { it.permissions.contains(permission) }
+
+    fun getVersionHashCode(user: UserEntity) = Objects.hash(
```

Comment:
this is a static function on the UserRole class. I think this can be a regular function on the UserEntity?